### DISCUSSION TITLE
documentation: add that @event.prevent is not conditional

### DISCUSSION THREAD

**Discussion by Cyrille37** [alpinejs/alpine#3618] [7. Bugs (archived)]
Hello
Thanks for this really great framework, it's a pleasure to code with <3

In the documentation [events#preventing-default](https://alpinejs.dev/essentials/events#preventing-default) is not clearly wrote that `@event.prevent` **is not conditional**. We may even think the opposite because of "..." in example ` <form @submit.prevent="...">...</form>`

A ref about: https://github.com/alpinejs/alpine/issues/344

Thanks & cheers :-)

**Comment by ekwoka** [comment]
I don't know it really needs a note for that.

If you want to conditionally do it, you use `$event.preventDefault()` in the handler. As noted: https://alpinejs.dev/essentials/events#accessing-the-event-object

How much of the alpine docs needs to just explain stuff that is javascript and nothing to do with Alpine?

**Reply by Cyrille37** [reply]
It's what I did ;-)

But after exploring that the "..." in `<form @submit.prevent="...">...</form>` does not mean that a conditional expression can be used, but only code execution.

**Reply by ekwoka** [reply]
sure. So you don't use `.prevent` there.

`.prevent` is a helper. It just handles something that is common for you

**Reply by Cyrille37** [reply]
I know, but when discovering Alpine.js it's not clearly wrote ... So we have to try if conditional expression works or not.

I think the documentation would not be damaged if this detail was added ;-)

**Reply by ekwoka** [reply]
You can make the PR.

I'm sure others might disagree with me on what amount of info should be in Alpine docs that isn't strictly Alpines behavior.

**Comment by Cyrille37** [comment]
I surrender :-)
